controllers/storageclient_controller.go

```diff
@@ -210,9 +217,25 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "Secret":
+			if eResource.Name == fmt.Sprintf("noobaa-remote-join-secret-%s", r.storageClient.Status.ConsumerID) {
+				data := map[string]string{}
+				if err := json.Unmarshal(eResource.Data, &data); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshall noobaa join secret data: %v", err)
+				}
```

Comment:
Can you please move the unmarshaling above the condition? All secrets will need unmarshaling of their data.

Reply:
The condition is required since hub cluster won't be having the join secret. Unmarshalling is only necessary if it's a remote storage client

Reply:
Also there might be future implementations where other secrets are fetched from provider.
We will need to differentiate between these too.